script.js

```diff
@@ -0,0 +1,107 @@
+class BookShop {
+  discountRateByBookNumber ;
+  topRatedBooks;
+  setDiscountRateByBookNumber (promotions) {
+    this.discountRateByBookNumber = {...promotions}
+  }
+
+  setTopRatedBooks (books){
+    this.topRatedBooks = books
+  }
+
+  priceSummation (data) {
+    let result = []
+    let transaction = {...data}
+
+    while (Object.keys(transaction).length !== 0 ) {
+      let sum = []
+      for (let key in transaction){
+        if (transaction[key] == 0){
+          delete transaction[key]
+        } else {
+          sum.push(transaction[key])
+          transaction[key] -= 1
+        }
+      }
+      sum.length > 1 ? result.push(sum.length*80*this.discountRateByBookNumber[sum.length]) : result.push(sum.length*80)
```

Comment:
use constant = 80 to avoid magic number

Reply:
Im not sure this fixed code is serve your point krub :)
` priceSummation (data) {
    let result = []
    const bookPrice = 80
    let transaction = {...data}`